src/audio/module_adapter/module_adapter.c

```diff
@@ -871,13 +871,14 @@ static inline int module_adapter_audio_stream_type_copy(struct comp_dev *dev)
 	struct comp_buffer __sparse_cache *source_c[PLATFORM_MAX_STREAMS];
 	struct comp_buffer __sparse_cache *sinks_c[PLATFORM_MAX_STREAMS];
 	struct processing_module *mod = comp_get_drvdata(dev);
+	struct module_data *md = &mod->priv;
 	struct list_item *blist;
 	uint32_t num_input_buffers, num_output_buffers;
 	int ret, i = 0;
 
 	/* handle special case of HOST/DAI type components */
 	if (dev->ipc_config.type == SOF_COMP_HOST || dev->ipc_config.type == SOF_COMP_DAI)
-		return module_process_legacy(mod, NULL, 0, NULL, 0);
+		return md->ops->process_audio_stream(mod, NULL, 0, NULL, 0);
```

Comment:
what about setting `md->state`?

Reply:
seems md->state is for ipc3 special, and copier_process does not depend on this, so there is no error reported in CI and QB.

Reply:
This is a change in behaviour. In other cases functions, that you inlined, really did nothing else apart from calling a method. And if it's really only needed for IPC3, removing those lines is more difficult to verify, because CI doesn't test IPC3 any more. It seems to me that `md->state` is something like a debugging flag, used to verify that we aren't running various operations in a wrong state, nothing else. Hitting such a wrong state would be an error, and I don't think we're hitting those errors ever, so maybe that `.state` could be removed completely.